bugbug/repository.py

```diff
@@ -310,27 +310,53 @@ def _hg_log(revs):
     return hg_log(HG, revs)
 
 
-def get_revs(hg, date_from=None):
-    # TODO: Retrieve all commits once calculate_experiences is faster and less memory hungry.
-    if date_from is not None:
-        pushdate = (date_from - relativedelta(months=12)).strftime("%Y-%m-%d")
-        rev_start = f"pushdate('>{pushdate}')"
-    else:
-        rev_start = 0
-
-    print(f"Getting revs from {rev_start} to tip...")
+def get_revs(hg):
+    print(f"Getting revs from 0 to tip...")
 
     args = hglib.util.cmdbuilder(
-        b"log",
-        template="{node}\n",
-        no_merges=True,
-        branch="central",
-        rev=f"{rev_start}:tip",
+        b"log", template="{node}\n", no_merges=True, branch="central", rev=f"0:tip"
     )
     x = hg.rawcommand(args)
     return x.splitlines()
 
 
+class exp_queue:
+    def __init__(self, start_day, maxlen, default):
+        self.list = deque([default] * maxlen, maxlen=maxlen)
+        self.start_day = start_day - (maxlen - 1)
+        self.default = default
+
+    @property
+    def last_day(self):
+        return self.start_day + (self.list.maxlen - 1)
+
+    def __getitem__(self, day):
+        assert day >= self.start_day, "Can't get a day from earlier than start day"
+
+        if day < 0:
+            return self.default
+
+        if day > self.last_day:
+            return self.list[-1]
+
+        return self.list[day - self.start_day]
+
+    def __setitem__(self, day, value):
+        if day == self.last_day:
+            self.list[day - self.start_day] = value
+        elif day > self.last_day:
+            last_val = self.list[-1]
+            self.list.extend(last_val for _ in range(self.last_day, day - 1))
```

Comment:
I'd say that adding last_val (N % self.maxlen) times instead of N times is better.
I'm not sure of that, just need to be checked but there is probably a modulo to put somewhere to improve speed.

Reply:
I think with 7740622 this should be good.

Reply:
ea29fe3 is improving even more, with a comment added in f6d77e1, please double check them!